pkg/tests/ossm/smcp_install_test.go

```diff
@@ -15,219 +15,244 @@
+			retry.UntilSuccess(t, func(t TestHelper) {
+				oc.AllResourcesDeleted(t,
+					meshNamespace,
+					assert.OutputContains("No resources found in",
+						"All resources deleted from namespace",
+						"Still waiting for resources to be deleted from namespace"))
+			})
+		})
+		t.NewSubTest("smcp_test_install_2.3").Run(func(t TestHelper) {
+			t.LogStep("Crete Namespace and Install SMCP v2.3")
+			oc.CreateNamespace(t, meshNamespace)
+			oc.ApplyTemplate(t, meshNamespace, smcpV23_template, vars)
+			if env.IsRosa() {
+				oc.PatchWithMerge(
+					t, meshNamespace,
+					fmt.Sprintf("smcp/%s", smcpName),
+					`{"spec":{"security":{"identity":{"type":"ThirdParty"}}}}`)
+			}
+			oc.WaitSMCPReady(t, meshNamespace, smcpName)
+			oc.ApplyString(t, meshNamespace, smmr)
+			t.LogStep("Check SMCP is Ready")
+			oc.WaitCondition(t, meshNamespace, "smcp", smcpName, "Ready")
+		})
+		t.NewSubTest("smcp_test_uninstall_2.3").Run(func(t TestHelper) {
+			t.LogStep("Delete SMCP v2.3 in namespace " + meshNamespace)
+			oc.DeleteFromString(t, meshNamespace, smmr)
+			oc.DeleteFromString(t, meshNamespace, template.Run(t, smcpV23_template, vars))
+			retry.UntilSuccess(t, func(t TestHelper) {
+				oc.AllResourcesDeleted(t,
+					meshNamespace,
+					assert.OutputContains("No resources found in",
+						"All resources deleted from namespace",
+						"Still waiting for resources to be deleted from namespace"))
+			})
+		})
+		t.NewSubTest("smcp_test_install_2.2").Run(func(t TestHelper) {
+			t.LogStep("Crete Namespace and Install SMCP v2.2")
+			oc.CreateNamespace(t, meshNamespace)
+			oc.ApplyTemplate(t, meshNamespace, smcpV22_template, vars)
+			if env.IsRosa() {
+				oc.PatchWithMerge(
+					t, meshNamespace,
+					fmt.Sprintf("smcp/%s", smcpName),
+					`{"spec":{"security":{"identity":{"type":"ThirdParty"}}}}`)
+			}
+			oc.WaitSMCPReady(t, meshNamespace, smcpName)
+			oc.ApplyString(t, meshNamespace, smmr)
+			t.LogStep("Check SMCP is Ready")
+			oc.WaitCondition(t, meshNamespace, "smcp", smcpName, "Ready")
+		})
+		t.NewSubTest("smcp_test_uninstall_2.2").Run(func(t TestHelper) {
+			t.LogStep("Delete SMCP v2.2 in namespace " + meshNamespace)
+			oc.DeleteFromString(t, meshNamespace, smmr)
+			oc.DeleteFromString(t, meshNamespace, template.Run(t, smcpV22_template, vars))
+			retry.UntilSuccess(t, func(t TestHelper) {
+				oc.AllResourcesDeleted(t,
+					meshNamespace,
+					assert.OutputContains("No resources found in",
+						"All resources deleted from namespace",
+						"Still waiting for resources to be deleted from namespace"))
+			})
+		})
+		t.NewSubTest("smcp_test_install_2.1").Run(func(t TestHelper) {
+			t.LogStep("Crete Namespace and Install SMCP v2.1")
+			oc.CreateNamespace(t, meshNamespace)
+			oc.ApplyTemplate(t, meshNamespace, smcpV21_template, vars)
+			if env.IsRosa() {
+				oc.PatchWithMerge(
+					t, meshNamespace,
+					fmt.Sprintf("smcp/%s", smcpName),
+					`{"spec":{"security":{"identity":{"type":"ThirdParty"}}}}`)
+			}
```

Comment:
Can we remove the patch and add this field to the SMCP yaml instead? This just makes both the code and the duration of the test longer.

Reply:
Fixed in another way, check the new commits

Reply:
Done, pending to test